waspc/src/Wasp/Lib.hs

```diff
@@ -33,40 +34,50 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
+  maybeAppSpec <- makeAppSpec waspDir options
+  case maybeAppSpec of
+    Left compileErrors -> return ([], compileErrors)
+    Right appSpec ->
+      case ASV.validateAppSpec appSpec of
+        [] -> do
+          (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
+          return (map show generatorWarnings, map show generatorErrors)
+        validationErrors -> do
+          return ([], map show validationErrors)
+
+makeAppSpec ::
```

Comment:
Ha this is the name you asked about :D!
Very nice extracting this!

`make` is common in Haskell, but I think it mostly implies construction, not so much a very complex operation, especially regarding side effects. It is often used for smart constructors. Here we are doing something very particular, which is reading wasp project from file and analyzing/compiling it. So I would name it more specifically. It could be `compileAppSpec`. Although that sounds like we are taking AppSpec and compiling it into smth. It could be `analyzeWaspProject`. Ha, maybe that is a nice one! Result is `AppSpec`, so that does make sense. Or it could be `compileWaspProject`. But that sounds like we do both analysis and generation, which we don't.
Btw `makeAppSpec` would be a good name for smth like `makeAppSpec :: [Decl] -> AppSpec`, which is what you are doing at the bottom of this function.
I think `analyzeWaspProject` is a good name.
I would only question if it should be returning `[CompileError]`, or smth more meaningful. Cause we know it is AnalyzeErrors really, right? Ok we do have that one "can't find a single *.wasp file", that complicates it a bit. But still it is mostly analyzer errors. Ok, this is just practical question then, do we need more control or not. Maybe we can stick with `CompileError` which is just a String really and then if needed we can always do one more extraction, one more function here, that will be the same thing really but returning AnalyzeErrors.

Reply:
Haha yeah thanks, agree `makeAppSpec` should be dumber than it is here. I originally was leaning `compileAppSpec` or `generateAppSpec` but felt they could be overloaded given those terms elsewhere. I'm good with `analyzeWaspProject`. Yeah I'd be inclined to maintain the same interface for errors as `compile` for now, but we can think on it. Thanks!